qcodes/tests/dataset/temporary_databases.py

```diff
@@ -20,7 +20,8 @@ def empty_temp_db():
     # create a temp database for testing
     with tempfile.TemporaryDirectory() as tmpdirname:
         qc.config["core"]["db_location"] = os.path.join(tmpdirname, 'temp.db')
-        qc.config["core"]["db_debug"] = True
+        if os.environ.get('QCODES_SQL_DEBUG'):
+            qc.config["core"]["db_debug"] = True
```

Comment:
Perhaps also this in order to ignore whatever the config setting is?
```suggestion
            qc.config["core"]["db_debug"] = True
        else:
            qc.config["core"]["db_debug"] = False
```